File: app/src/main/java/com/lucciola/haskellanywhereforandroid/uiaction/UIAction.java
package com.lucciola.haskellanywhereforandroid.uiaction;

import android.os.AsyncTask;

import com.lucciola.haskellanywhereforandroid.View;
import com.lucciola.haskellanywhereforandroid.ProgramManager;

/**
 * Created by hiden on 2017/06/17.
 */

public abstract class UIAction extends AsyncTask<Void, Void, String> {
    protected View view;
    protected String inputString;
    protected String documentString;
    protected ProgramManager manager;

    public UIAction(ProgramManager arg0, View arg2) {
        this.manager = arg0;
        this.view = arg2;
    }

}

File: app/src/main/kotlin/com/lucciola/haskellanywhereforandroidkt/uiaction/UIAction.kt
package com.lucciola.haskellanywhereforandroidkt.uiaction

import android.os.AsyncTask

import com.lucciola.haskellanywhereforandroidkt.View
import com.lucciola.haskellanywhereforandroidkt.ProgramManager

/**
 * Created by hiden on 2017/06/17.
 */

abstract class UIAction(protected var manager: ProgramManager, protected var view: View) : AsyncTask<Void, Void, String>() {
    protected var inputString: String? = null
    protected var documentString: String? = null

}
